<?xml version="1.0"?>
<launch>
  <arg name="shutter_time" default="200" />
  <arg name="idle_time" default="800"/>
  <arg name="container_size" default="10"/>
  <arg name="lag_compensation" default="0.001"/>
  <arg name="topic" default="laser_scan"/>
  <arg name="frame" default="scanCONTROL_2900-50_scanner_laser_link"/>
  <arg name="dense" default="false"/>
  <arg name="field_left" default="0.0"/>
  <arg name="field_right" default="0.0"/>
  <arg name="field_far" default="0.0"/>
  <arg name="field_near" default="0.0"/>
  <arg name="serial_number" default="218020023"/> <!-- 216090134 -->
  <arg name="path_to_device_properties" default="/home/syn/ros_ws/src/microepsilon_scancontrol/misc/"/> <!-- "/opt/scanCONTROL/" -->
  <arg name="ns" default="arm"/>
  <node ns = "$(arg ns)" name="microepsilon_scancontrol_node" pkg="microepsilon_scancontrol" type="microepsilon_scancontrol_node" cwd="node" respawn="true" output="screen">
    <param name="shutter_time" type="int" value="$(arg shutter_time)"/>
    <param name="idle_time" type="int" value="$(arg idle_time)"/>
    <param name="container_size" type="int" value="$(arg container_size)"/>
    <param name="lag_compensation" type="double" value="$(arg lag_compensation)"/>
    <param name="topic" type="string" value="$(arg topic)"/>
    <param name="frame" type="string" value="$(arg frame)"/>
    <param name="dense" type="bool" value="$(arg dense)"/>
    <param name="field_left" type="double" value="$(arg field_left)"/>
    <param name="field_right" type="double" value="$(arg field_right)"/>
    <param name="field_far" type="double" value="$(arg field_far)"/>
    <param name="field_near" type="double" value="$(arg field_near)"/>
    <param name="serial_number" type="string" value="$(arg serial_number)"/>
    <param name="path_to_device_properties" type="string" value="$(arg path_to_device_properties)"/>
  </node>
</launch>
